install/gcp/installer/util/gcloud.py

```diff
@@ -54,6 +54,16 @@ def get_gcloud_info():
     return project_id, authed_user, is_devshell
 
 
+def set_project_id(self):
+    """Set the project."""
+    print('Setting Project %s' % self.config.project_id)
+    return_code, out, err = utils.run_command(
+        ['gcloud', 'config', 'set', 'project', self.config.project_id])
```

Comment:
this changes state on the running user's machine. can we append --project flag to the gcloud deployment-manager call instead?

Reply:
The current deployment-manager call expects the installation project to be the current project id in context of the running user. It was much easier to simply changing the running user's context than to add more logic later. It can be done the way you're asking, just requires more changes and this flow felt more natural.

Reply:
Do note that this behaviour is inconsistent with gcloud binary (which is ok, but imo better to be consistent as that's what users expect working with gcp). 

gcloud some command --project=foo does not change state of project on the machine

Reply:
I can agree to that. A TODO to remove dep on gcloud config set project X, and add the --project arg to each related gcloud call.